.ci/test_r_package.sh

```diff
@@ -130,7 +155,7 @@ if grep -q -R "WARNING" "$LOG_FILE_NAME"; then
     exit -1
 fi
 
-ALLOWED_CHECK_NOTES=3
+ALLOWED_CHECK_NOTES=4
```

Comment:
Can you please clarify which one is +1?

Reply:
oh! I have been developing this for about 3 months, very possible that this can go back down to 3. I might have changed it to 4 at some point to keep up with changes in the repo, for example for the issue fixed in #3181 

when CI finishes building for the commit I just pushed, I'll go look through the logs and hopefully this can go back to 3.

Reply:
ok it seems like this is outdated, and actually can be reduced to `2` in most cases!

```text
CMake builds:

* ubuntu + gcc + R 3.6: 2 NOTEs
* ubuntu + gcc + R 4.0: 2 NOTEs
* ubuntu + clang + R 3.6: 2 NOTEs
* ubuntu + clang +  R 4.0: 2 NOTEs
* mac + gcc + R 3.6: 2 NOTEs
* mac + gcc + R 4.0: 2 NOTEs
* mac + clang + R 3.6: 2 NOTEs
* mac + clang + R 4.0: 2 NOTEs
* Windows + MINGW + R 3.6: 3 NOTEs
* Windows + MINGW + R 4.0: 3 NOTEs

CRAN:

* ubuntu + R 4.0: 3 NOTEs
* mac + R 4.0: 2 NOTEs
```

The 2 NOTEs are not problems. The extra one on ubuntu is thiis:

```
* checking installed package size ... NOTE
  installed size is 37.1Mb
  sub-directories of 1Mb or more:
    libs  36.5Mb
```

It's not being caused by this PR, and it is weird to me that it only shows up on this one build for Linux (and on all the Windows builds). I think it is probably a result of differences in how the library is linked...something we can try to solve in a future PR.

I pushed a commit which adds this to `test_r_package.sh`

```shell
if [[ $OS_NAME == "linux" ]] && [[ $R_BUILD_TYPE == "cran" ]]; then
    ALLOWED_CHECK_NOTES=3
else
    ALLOWED_CHECK_NOTES=2
fi
```